src/Microsoft.ML.Transforms/PermutationFeatureImportance.cs

```diff
@@ -9,19 +9,19 @@
 using System.Text;
 using Microsoft.Data.DataView;
 using Microsoft.ML.Data;
-using Microsoft.ML.Internal.Internallearn;
 using Microsoft.ML.Internal.Utilities;
 using Microsoft.ML.Model;
 
 namespace Microsoft.ML.Transforms
 {
-    internal static class PermutationFeatureImportance<TModel, TMetric, TResult> where TResult : MetricsStatisticsBase<TMetric>, new()
+    internal static class PermutationFeatureImportance<TModel, TMetric, TResult> where TResult : IMetricsStatistics<TMetric>
```

Comment:
Is it not possible to keep the `new` constraint?  That seems cleaner than passing in a `Func<TResult>` IMO.

Reply:
I disagree in this case, since I do not want this thing to be publicly instantiable, and the `new()` constraint required that it have a *public* empty constructor, which I do not consider necessary for the purpose. (The public is the key problem here.)

So I had a choice: do I internally have this little factory method, or do I have a publicly accessible type just to enable this infrastructure. Seemed like a really straightforward choice to me.

Reply:
Ok - the consideration I was missing here was that the type needed to have a public ctor. This approach seems right then.